Makefile.am

```diff
@@ -31,7 +31,7 @@ AM_CPPFLAGS += -DDATADIR='"$(datadir)"' -DLIBEXECDIR='"$(libexecdir)"' \
 	-DOSTREE_COMPILATION \
 	-DG_LOG_DOMAIN=\"OSTree\" \
   -DOSTREE_GITREV='"$(OSTREE_GITREV)"' \
-	-DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40 '-DGLIB_VERSION_MAX_ALLOWED=G_ENCODE_VERSION(2,50)' \
+	-DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_44 '-DGLIB_VERSION_MAX_ALLOWED=G_ENCODE_VERSION(2,50)' \
```

Comment:
Potentially worth refactoring this as a prep commit.  It looks like Debian stable has 2.58 and oldstable has 2.50, so we can almost certainly go to at least 2.50.

And I bet there's other cruft we could drop then, like our backported autocleanup macros.

Reply:
Sure, do you want to go for 2.50?

Reply:
SGTM!

Reply:
Hmm, bumping to 2.50 results in a pile of deprecation warnings for `GSimpleAsyncResult` usage, which was deprecated in 2.46 in favor of `GTask`. See https://developer.gnome.org/gio/stable/GSimpleAsyncResult.html. I'm going to leave this at 2.44 as I don't really want to go refactor all of that as part of this work.